### ISSUE TITLE
Assumes everything that's .clj is a file, even when it's a directory

### ISSUE THREAD

**Issue by SevereOverfl0w** [borkdude/clj-kondo#394]
**version**

clj-kondo v2019.07.31-alpha

**platform**

Linux - GraalVM

**problem**

When there is a directory ending in  ".clj" then kondo assumes it's a file and tries to read it.

**repro**

```
/tmp/skfjkjdsf                                                                                                                                         0.04 17:49:37
❯ mkdir src/                   

/tmp/skfjkjdsf                                                                                                                                         0.03 17:49:43
❯ mkdir src/bar.clj            

/tmp/skfjkjdsf                                                                                                                                         0.02 17:49:48
❯ clj-kondo --lint src
Unexpected error. Please report an issue.
java.io.IOException: Is a directory
        at com.oracle.svm.core.posix.PosixUtils.newIOExceptionWithLastError(PosixUtils.java:216)
        at com.oracle.svm.core.posix.PosixUtils.readBytes(PosixUtils.java:339)
        at java.io.FileInputStream.readBytes(FileInputStream.java:522)
        at java.io.FileInputStream.read(FileInputStream.java:255)
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
        at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
        at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
        at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
        at java.io.InputStreamReader.read(InputStreamReader.java:184)
        at java.io.BufferedReader.fill(BufferedReader.java:161)
        at java.io.BufferedReader.read1(BufferedReader.java:212)
        at java.io.BufferedReader.read(BufferedReader.java:286)
        at java.io.Reader.read(Reader.java:140)
        at clojure.java.io$fn__11538.invokeStatic(io.clj:337)
        at clojure.java.io$fn__11538.invoke(io.clj:334)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at clojure.java.io$copy.invokeStatic(io.clj:406)
        at clojure.java.io$copy.doInvoke(io.clj:391)
        at clojure.lang.RestFn.invoke(RestFn.java:425)
        at clojure.core$slurp.invokeStatic(core.clj:6951)
        at clojure.core$slurp.doInvoke(core.clj:6942)
        at clojure.lang.RestFn.invoke(RestFn.java:410)
        at clj_kondo.impl.core$sources_from_dir$fn__4332.invoke(core.clj:109)
        at clojure.core$keep$fn__8559.invoke(core.clj:7341)
        at clojure.lang.LazySeq.sval(LazySeq.java:42)
        at clojure.lang.LazySeq.seq(LazySeq.java:58)
        at clojure.lang.RT.seq(RT.java:535)
        at clojure.core$seq__5402.invokeStatic(core.clj:137)
        at clojure.core$map$fn__5866.invoke(core.clj:2746)
        at clojure.lang.LazySeq.sval(LazySeq.java:42)
        at clojure.lang.LazySeq.seq(LazySeq.java:58)
        at clojure.lang.RT.seq(RT.java:535)
        at clojure.core$seq__5402.invokeStatic(core.clj:137)
        at clojure.core$dorun.invokeStatic(core.clj:3133)
        at clojure.core$doall.invokeStatic(core.clj:3148)
        at clj_kondo.core$run_BANG_.invokeStatic(core.clj:90)
        at clj_kondo.main$main.invokeStatic(main.clj:74)
        at clj_kondo.main$main.doInvoke(main.clj:67)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:665)
        at clj_kondo.main$_main$fn__4451.invoke(main.clj:93)
        at clj_kondo.main$_main.invokeStatic(main.clj:93)
        at clj_kondo.main$_main.doInvoke(main.clj:91)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clj_kondo.main.main(Unknown Source)
```

**expected behavior**

Ignores the directory

**actual behavior**

:boom:

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/f8b0de0f76b7c78d648c93e672e3eb2caf70e45d.

Thanks @mk for hacking together on Heart of Clojure! :-)

**Comment by borkdude** [comment]
I'll close this issue once it's released.
